test/index.ts

```diff
@@ -557,23 +557,55 @@ function testNonEmptyArray() {
 }
 
 function testReadonlyKeys() {
-  type T = { readonly a: number; b: string };
-
-  type Actual = ReadonlyKeys<T>;
-
-  type Expected = "a";
-
-  type Test = Assert<IsExact<Actual, Expected>>;
+  type cases = [
+    // @ts-expect-error primitives not allowed
```

Comment:
The description doesn't make much sense, because you still return something (and it's not `never`), therefore it should be updated for the majority of the primitives?